test/stringio/test_stringio.rb

```diff
@@ -1010,6 +1010,54 @@ def test_chilled_string_string_set
     end
   end
 
+  def test_to_s
+    content = "Hello, StringIO!"
+    stringio = StringIO.new(content)
+
+    assert_equal content, stringio.to_s
+    assert_equal content, stringio.string
+    assert_equal stringio.string, stringio.to_s
+
+    stringio.pos = 7
+    assert_equal content, stringio.to_s
+
+    stringio.close
+    assert_equal content, stringio.to_s
+
+    empty_stringio = StringIO.new
+    assert_equal "", empty_stringio.to_s
+
+    stringio_with_encoding = StringIO.new(content.encode("UTF-16LE"))
+    assert_equal content.encode("UTF-16LE"), stringio_with_encoding.to_s
+    assert_equal "UTF-16LE", stringio_with_encoding.to_s.encoding.name
+  end
+
+  def test_to_s_with_interpolation
+    content = "Hello, StringIO!"
+    stringio = StringIO.new(content)
+
+    assert_equal "Content: #{content}", "Content: #{stringio}"
+
+    assert_equal "Before #{content} After", "Before #{stringio} After"
+
+    empty_stringio = StringIO.new
+    assert_equal "Empty: ", "Empty: #{empty_stringio}"
+
+    utf16_content = content.encode("UTF-16LE")
+    stringio_with_encoding = StringIO.new(utf16_content)
+    assert_equal "UTF-16LE: #{utf16_content.encode('UTF-8')}", "UTF-16LE: #{stringio_with_encoding.to_s.encode('UTF-8')}"
+
+    assert_equal "UTF-16LE", stringio_with_encoding.to_s.encoding.name
+
+    stringio.pos = 7
+    assert_equal "Moved: #{content}", "Moved: #{stringio}"
+
+    stringio.close
+    assert_equal "Closed: #{content}", "Closed: #{stringio}"
+
+    assert_equal "Start #{content} Middle #{empty_stringio} End", "Start #{stringio} Middle #{empty_stringio} End"
+  end
+
```

Comment:
The interpolation is not a feature of `StringIO`.
This does not make sense here.